src/hardware/encoder.cpp

```diff
@@ -714,35 +726,37 @@ double Encoder::getTemp() {
 }
 
 
-// Gets the raw revolutions from the motor
-double Encoder::getRawRev() {
+// Gets the raw revolutions from the motor in range [-258 ... +257]
+int16_t Encoder::getRawRev() {
 
     // Create an accumulator for the raw data and converted data
     uint16_t rawData;
-    int16_t convertedData;
 
     // Loop continuously until there is no error
     while (readRegister(ENCODER_ANGLE_REV_REG, rawData) != NO_ERROR);
 
     // Delete the first 7 bits, they are not needed
     rawData = (rawData & (DELETE_7_BITS));
 
-    // Copy the raw data over, now that it's almost converted
-    convertedData = rawData;
-
     // Check if the value is negative, if so it needs 512 subtracted from it
-    if (convertedData & CHECK_BIT_9) {
-        convertedData -= 512;
+    if (rawData & CHECK_BIT_9) {
+        rawData -= CHANGE_UNIT_TO_INT_9;
     }
 
     // Return the angle measurement
-    return ((double)convertedData);
+    return (int16_t)rawData;
 }
 
 
 // Gets the revolutions of the motor
 double Encoder::getRev() {
-    return (getRawRev() - startupRevOffset);
+    int16_t rawRevNow = getRawRev();
+    if ((lastRawRev > 0) && (rawRevNow < 0)) // overwlow
+        revolutions++;
+    else if ((lastRawRev < 0) && (rawRevNow > 0)) // borrow
+        revolutions--;
+    lastRawRev = rawRevNow;
+    return (double)revolutions * 512 + rawRevNow - startupRevOffset;
 }
```

Comment:
Note that this function requires a different average. The issue is that when the angle overflowing into a rotation, the angle would be all messed up. However, then encoder still needs averaged as the readings aren't stable enough otherwise

Reply:
@CAP1Sup  Do you mean bounce in position where changing shaft revolutions?

Reply:
What I'm referring to is that the readings spike due to the angle average going from 360 to 0 or vice versa. This causes delay in accuracy of the encoder readings. I fixed it now by using an absolute angle avg. I originally made the same mistake.